lib/drivers/medtronic600/medtronic600Driver.js

```diff
@@ -1325,18 +1325,29 @@ class BasicNgpParametersCommand extends TransmitPacketRequest {
 
 class BolusWizardCarbRatiosResponse extends TransmitPacketResponse {
   get ratios() {
-    const ratios = [];
-    // Bytes 0x03 and 0x04 are a CCITT checksum of the ratios bytes.
-    const numItems = this.comDPayload[0x05];
+    const readRatios = (offset, factor) => {
+      const ratios = [];
+      // Bytes 0x03 and 0x04 are a CCITT checksum of the ratios bytes.
+      const numItems = this.comDPayload[0x05];
 
-    for (let i = 0; i < numItems; i++) {
-      const amount = (this.comDPayload.readUInt32BE(0x06 + (i * 9))) / 10;
-      // There is another UInt32BE after the amount, which is the exchange ratio (in 1000 units per exchange).
+      for (let i = 0; i < numItems; i++) {
+        const amount = (this.comDPayload.readUInt32BE(offset + (i * 9))) / factor;
 
-      ratios.push({
-        start: this.comDPayload[0x0E + (i * 9)] * 30 * sundial.MIN_TO_MSEC,
-        amount,
-      });
+        ratios.push({
+          start: this.comDPayload[0x0E + (i * 9)] * 30 * sundial.MIN_TO_MSEC,
+          amount,
+        });
+      }
+      return ratios;
+    };
+
+    let ratios = readRatios(0x06, 10);
+    if (ratios.every(item => item.amount === 0)) {
+      // if carb ratios are all zero, they are in exchanges instead of
+      // grams and stored in a different UInt32BE
+      // Since pumpSettings doesn't support `exchanges` as a unit, convert all
+      // exchanges to grams.
+      ratios = readRatios(0x0A, 1000 * 15.0);
```

Comment:
Are exchanges always 15g? I think I've seen instances in the past where it was 10g? Isn't it essentially up to the user?

Reply:
I based it on the the pump settings history: https://github.com/tidepool-org/uploader/blob/develop/lib/drivers/medtronic600/NGPHistoryParser.js#L1384-L1390

I guess the conversion factor stays the same per pump model? I.e., you choose either exchanges or grams, you don't choose the actual conversion factor.